.gitlab-ci.yml

```diff
@@ -348,22 +348,33 @@ TF_Settings::Windows::build:
 TF_Settings::Web::build:
   stage: build
   when: always
+  image: registry.ultrahaptics.com/ooh/infrastructure/windows-touchfree-toolchain:latest
   script:
     - npm -v
     - npm install -w TF_Settings_Web
     - npm ls
     - cd ./TF_Settings_Web
-    - npm run build
-    - npx generate-license-file --input ./package.json --output ./dist/ThirdPartyLicenses.txt --eol posix
+    - npm run tauri build
+    # npm workspaces cause the Web Settings node modules to be installed in the root folder so need to copy the Web Settings' package.json up a level
```

Comment:
Why do we need to do this? Won't it overwrite the workspaces package.json? 🤔

Reply:
We've already built the app and everything so it's fine to overwrite the other package.json as it's being used again (and docker image so won't affect other runs). 

We need to do it as TF_Settings_Web has no node modules (they get installed at the root directory) so generating the license gives you an empty file (has been broken since we did the workspaces change). For the license to correctly populate we need the web setting's package.json and the node modules at the same level so need to copy the package.json up a level to the root directory.
I added a comment trying to explain this but if you don't think it's clear then let me know what I missed

Also added a check to fail the job fi the license file doesn't contain React (as it definitely should if it populated correctly)

Reply:
It's possible for a node_modules to get generated in the TF_Settings_Web directory - it will miss licenses from those with this setup. It's ok for now if that's not happening (I believe it happens with some kind of situation with ambiguous dependencies) but maybe it's worth putting a check in place to alert us if that starts happening? Something really basic like the job checks if a `node_modules` exists in TF_Settings_Web and warns us it's time to think about this.